canonical-web-and-design/maas-ui
# feat(ui): make the machine test and fieldless forms reusable

## Done

- Make the machine test and fieldless forms reusable by controllers.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the machine list.
- Check that you can still use the take action menu to test a machine.
- Check that you can still use the take action menu to preform an action without a form e.g. turn on, lock etc.

## Fixes

Fixes: canonical-web-and-design/app-tribe#614.
